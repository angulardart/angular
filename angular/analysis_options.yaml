# This is a copy of ../analysis_options.yaml, with implicit-casts: false, and an "exclude: ...".

analyzer:
  strong-mode:
    implicit-casts: false
  exclude:
    - "tools/analyzer_plugin/bin/plugin.dart"
  errors:
    unused_element: error
    unused_import: error
    unused_local_variable: error
    dead_code: error

    # Allow importing .template.dart files without an analyzer error.
    uri_has_not_been_generated: ignore

    # The noise of deprecations isn't worth the effort.
    deprecated_member_use_from_same_package: ignore
linter:
  rules:
    - avoid_empty_else
    - avoid_init_to_null
    - avoid_null_checks_in_equality_operators
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - control_flow_in_finally
    - directives_ordering
    - empty_catches
    - empty_constructor_bodies
    - empty_statements
    - hash_and_equals
    - iterable_contains_unrelated_type
    - library_names
    - library_prefixes
    - list_remove_unrelated_type
    - only_throw_errors
    - overridden_fields
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - prefer_final_fields
    - prefer_is_not_empty
    - slash_for_doc_comments
    - test_types_in_equals
    - throw_in_finally
    - type_init_formals
    - unnecessary_brace_in_string_interps
    - unrelated_type_equality_checks
    - valid_regexps
