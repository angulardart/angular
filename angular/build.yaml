targets:
  $default:
    builders:
      angular:
        generate_for:
          exclude:
            - "lib/src/build.dart"
            - "lib/src/compiler/**"
            - "lib/src/source_gen/**"

builders:
  angular:
    import: "package:angular/src/build.dart"
    builder_factories:
      - templatePlaceholder
      - templateCompiler
      - stylesheetCompiler
    auto_apply: dependents
    applies_builders:
      - "angular|placeholder_cleanup"
      - "angular|component_source_cleanup"
    # See https://github.com/dart-lang/angular/issues/988.
    is_optional: true
    required_inputs:
      - ".css"
    build_extensions:
      .css:
        - ".css.dart"
        - ".css.shim.dart"
      .dart:
        - ".template.dart"
post_process_builders:
  placeholder_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "placeholderCleanup"
  component_source_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "componentSourceCleanup"
    defaults:
      generate_for: ["lib/**"]
      release_options:
        enabled: true
