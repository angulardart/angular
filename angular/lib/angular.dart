/// The primary library for the [AngularDart web framework][AngularDart].
///
/// Import this library as follows:
///
/// ```
/// import 'package:angular/angular.dart';
/// ```
///
/// For help using this library, see the AngularDart documentation:
///
/// * [AngularDart guide][]
/// * [AngularDart cheat sheet][cheatsheet]
///
/// [AngularDart]: https://webdev.dartlang.org/angular
/// [AngularDart guide]: https://webdev.dartlang.org/angular/guide
/// [cheatsheet]: https://webdev.dartlang.org/angular/cheatsheet

library angular;

export 'core.dart';
export 'src/common/common_directives.dart';
export 'src/common/directives.dart';
export 'src/common/pipes.dart';
export 'src/core/angular_entrypoint.dart' show AngularEntrypoint;
export 'src/core/application_tokens.dart' hide APP_ID_RANDOM_PROVIDER;
export 'src/platform/bootstrap.dart';
export 'src/platform/dom/dom_tokens.dart';
export 'src/platform/dom/events/event_manager.dart' show EventManagerPlugin;
