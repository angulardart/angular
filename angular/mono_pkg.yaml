dart:
- stable
- dev

stages:
- analyze:
  - dartanalyzer: --fatal-warnings .
