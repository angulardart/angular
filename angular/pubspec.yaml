name: angular
version: 5.3.0
author: Dart Team <web@dartlang.org>
description: Fast and productive web framework
homepage: https://webdev.dartlang.org/angular
documentation: https://webdev.dartlang.org/angular/api
environment:
  sdk: ">=2.2.1-dev.3.0 <3.0.0"

dependencies:
  # Runtime (or runtime *and* the compiler).
  collection: ^1.14.5
  intl: '>=0.13.0 <0.16.0'
  js: ^0.6.0
  logging: '>=0.9.0 <0.12.0'
  meta: ^1.1.4
  stack_trace: ^1.9.1

  # Compiler. Eventually we want to move these to angular_compiler.
  analyzer: ^0.36.2

  # It is extremely important these two dependencies stay pinned to current.
  angular_ast: 0.5.9
  angular_compiler: 0.4.3
  ##########################################################################

  build: '>=0.12.7 <2.0.0'
  build_config: '>=0.2.6 <0.5.0' # Runtime dependency with no import
  code_builder: '^3.0.1'
  csslib: '>=0.14.5 <0.16.0'
  path: ^1.6.1
  source_gen: ">=0.9.1 <0.10.0"
  source_span: ^1.4.0

# === vvv REMOVE WHEN PUBLISHING vvv ===
dependency_overrides:
  angular_ast:
      path: ../angular_ast
  angular_compiler:
      path: ../angular_compiler
# === ^^^ REMOVE WHEN PUBLISHING ^^^ ===
