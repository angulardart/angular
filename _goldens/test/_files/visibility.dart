import 'package:angular/angular.dart';

abstract class Dependency {}

@Component(
  selector: 'dependent',
  template: '',
)
class Dependent {
  Dependent(Dependency _);
}

@Component(
  selector: 'dependency-with-dependent-in-view',
  template: '<dependent></dependent>',
  directives: [Dependent],
  providers: [
    Provider(Dependency, useExisting: DependencyWithDependentInView),
  ],
)
class DependencyWithDependentInView implements Dependency {}

@Component(
  selector: 'dependency-with-content',
  template: '<ng-content></ng-content>',
  providers: [
    Provider(Dependency, useExisting: DependencyWithContent),
  ],
)
class DependencyWithContent implements Dependency {}

@Component(
  selector: 'dependency-and-dependent-in-view',
  template: '''
    <dependency>
      <dependent></dependent>
    </dependency>
  ''',
  directives: [Dependent],
)
class DependencyAndDependentInView {}

@Component(
  selector: 'has-visibility-all',
  template: '',
  visibility: Visibility.all,
)
class HasVisibilityAll {}

@Component(
  selector: 'has-visibility-local',
  template: '',
)
class HasVisibilityLocal {}
