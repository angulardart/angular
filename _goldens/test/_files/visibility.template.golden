// **************************************************************************
// Generator: AngularDart Compiler
// **************************************************************************

import 'visibility.dart';
export 'visibility.dart';
import 'package:angular/src/di/reflector.dart' as _ngRef;
import 'package:angular/angular.template.dart' as _ref0;
import 'package:angular/src/core/linker/views/component_view.dart' as import0;
import 'visibility.dart' as import1;
import 'package:angular/src/core/linker/style_encapsulation.dart' as import2;
import 'package:angular/src/core/linker/views/view.dart' as import3;
import 'package:angular/src/core/change_detection/change_detection.dart';
import 'dart:html' as import5;
import 'package:angular/src/runtime.dart' as import6;
import 'package:angular/angular.dart';
import 'package:angular/src/core/linker/views/host_view.dart' as import8;
import 'package:angular/src/di/errors.dart' as import9;
import 'package:angular/src/runtime/dom_helpers.dart' as import10;

final List<dynamic> styles$Dependent = const [];

// View for component Dependent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class ViewDependent0 extends import0.ComponentView<import1.Dependent> {
  static import2.ComponentStyles _componentStyles;
  ViewDependent0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('dependent');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/visibility.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$Dependent, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.Dependent> _DependentNgFactory = const ComponentFactory('dependent', viewFactory_DependentHost0);
ComponentFactory<import1.Dependent> get DependentNgFactory {
  return _DependentNgFactory;
}

final List<dynamic> styles$DependentHost = const [];

// Host for component Dependent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class _ViewDependentHost0 extends import8.HostView<import1.Dependent> {
  @override
  void build() {
    // Child component Dependent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this.componentView = ViewDependent0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = (import6.isDevMode
        ? import9.debugInjectorWrap(import1.Dependent, () {
            return import1.Dependent(this.injectorGet(import1.Dependency, this.parentIndex));
          })
        : import1.Dependent(this.injectorGet(import1.Dependency, this.parentIndex)));
    this.initRootNode(_el_0);
  }
}

import8.HostView<import1.Dependent> viewFactory_DependentHost0() {
  return _ViewDependentHost0();
}

final List<dynamic> styles$DependencyWithDependentInView = const [];

// View for component DependencyWithDependentInView in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class ViewDependencyWithDependentInView0 extends import0.ComponentView<import1.DependencyWithDependentInView> {
  ViewDependent0 _compView_0;
  import1.Dependent _Dependent_0_5;
  static import2.ComponentStyles _componentStyles;
  ViewDependencyWithDependentInView0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('dependency-with-dependent-in-view');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/visibility.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    // Child component Dependent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this._compView_0 = ViewDependent0(this, 0);
    final _el_0 = this._compView_0.rootElement;
    parentRenderNode.append(_el_0);
    this._Dependent_0_5 = (import6.isDevMode
        ? import9.debugInjectorWrap(import1.Dependent, () {
            return import1.Dependent(this.parentView.injectorGet(import1.Dependency, this.parentIndex));
          })
        : import1.Dependent(this.parentView.injectorGet(import1.Dependency, this.parentIndex)));
    this._compView_0.create(this._Dependent_0_5);
  }

  @override
  void detectChangesInternal() {
    this._compView_0.detectChanges();
  }

  @override
  void destroyInternal() {
    this._compView_0.destroyInternalState();
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$DependencyWithDependentInView, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.DependencyWithDependentInView> _DependencyWithDependentInViewNgFactory = const ComponentFactory('dependency-with-dependent-in-view', viewFactory_DependencyWithDependentInViewHost0);
ComponentFactory<import1.DependencyWithDependentInView> get DependencyWithDependentInViewNgFactory {
  return _DependencyWithDependentInViewNgFactory;
}

final List<dynamic> styles$DependencyWithDependentInViewHost = const [];

// Host for component DependencyWithDependentInView in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class _ViewDependencyWithDependentInViewHost0 extends import8.HostView<import1.DependencyWithDependentInView> {
  @override
  void build() {
    // Child component DependencyWithDependentInView in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this.componentView = ViewDependencyWithDependentInView0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.DependencyWithDependentInView();
    this.initRootNode(_el_0);
  }

  @override
  dynamic injectorGetInternal(dynamic token, int nodeIndex, dynamic notFoundResult) {
    if ((identical(token, import1.Dependency) && (0 == nodeIndex))) {
      return this.component;
    }
    return notFoundResult;
  }
}

import8.HostView<import1.DependencyWithDependentInView> viewFactory_DependencyWithDependentInViewHost0() {
  return _ViewDependencyWithDependentInViewHost0();
}

final List<dynamic> styles$DependencyWithContent = const [];

// View for component DependencyWithContent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class ViewDependencyWithContent0 extends import0.ComponentView<import1.DependencyWithContent> {
  static import2.ComponentStyles _componentStyles;
  ViewDependencyWithContent0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('dependency-with-content');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/visibility.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    this.project(parentRenderNode, 0);
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$DependencyWithContent, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.DependencyWithContent> _DependencyWithContentNgFactory = const ComponentFactory('dependency-with-content', viewFactory_DependencyWithContentHost0);
ComponentFactory<import1.DependencyWithContent> get DependencyWithContentNgFactory {
  return _DependencyWithContentNgFactory;
}

final List<dynamic> styles$DependencyWithContentHost = const [];

// Host for component DependencyWithContent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class _ViewDependencyWithContentHost0 extends import8.HostView<import1.DependencyWithContent> {
  @override
  void build() {
    // Child component DependencyWithContent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this.componentView = ViewDependencyWithContent0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.DependencyWithContent();
    this.initRootNode(_el_0);
  }

  @override
  dynamic injectorGetInternal(dynamic token, int nodeIndex, dynamic notFoundResult) {
    if ((identical(token, import1.Dependency) && (0 == nodeIndex))) {
      return this.component;
    }
    return notFoundResult;
  }
}

import8.HostView<import1.DependencyWithContent> viewFactory_DependencyWithContentHost0() {
  return _ViewDependencyWithContentHost0();
}

final List<dynamic> styles$DependencyAndDependentInView = const [];

// View for component DependencyAndDependentInView in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class ViewDependencyAndDependentInView0 extends import0.ComponentView<import1.DependencyAndDependentInView> {
  ViewDependent0 _compView_1;
  import1.Dependent _Dependent_1_5;
  static import2.ComponentStyles _componentStyles;
  ViewDependencyAndDependentInView0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('dependency-and-dependent-in-view');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/visibility.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
    final doc = import5.document;
    final _el_0 = import10.appendElement(doc, parentRenderNode, 'dependency');
    // Child component Dependent in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this._compView_1 = ViewDependent0(this, 1);
    final _el_1 = this._compView_1.rootElement;
    _el_0.append(_el_1);
    this._Dependent_1_5 = (import6.isDevMode
        ? import9.debugInjectorWrap(import1.Dependent, () {
            return import1.Dependent(this.parentView.injectorGet(import1.Dependency, this.parentIndex));
          })
        : import1.Dependent(this.parentView.injectorGet(import1.Dependency, this.parentIndex)));
    this._compView_1.create(this._Dependent_1_5);
  }

  @override
  void detectChangesInternal() {
    this._compView_1.detectChanges();
  }

  @override
  void destroyInternal() {
    this._compView_1.destroyInternalState();
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$DependencyAndDependentInView, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.DependencyAndDependentInView> _DependencyAndDependentInViewNgFactory = const ComponentFactory('dependency-and-dependent-in-view', viewFactory_DependencyAndDependentInViewHost0);
ComponentFactory<import1.DependencyAndDependentInView> get DependencyAndDependentInViewNgFactory {
  return _DependencyAndDependentInViewNgFactory;
}

final List<dynamic> styles$DependencyAndDependentInViewHost = const [];

// Host for component DependencyAndDependentInView in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class _ViewDependencyAndDependentInViewHost0 extends import8.HostView<import1.DependencyAndDependentInView> {
  @override
  void build() {
    // Child component DependencyAndDependentInView in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this.componentView = ViewDependencyAndDependentInView0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.DependencyAndDependentInView();
    this.initRootNode(_el_0);
  }
}

import8.HostView<import1.DependencyAndDependentInView> viewFactory_DependencyAndDependentInViewHost0() {
  return _ViewDependencyAndDependentInViewHost0();
}

final List<dynamic> styles$HasVisibilityAll = const [];

// View for component HasVisibilityAll in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class ViewHasVisibilityAll0 extends import0.ComponentView<import1.HasVisibilityAll> {
  static import2.ComponentStyles _componentStyles;
  ViewHasVisibilityAll0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-visibility-all');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/visibility.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasVisibilityAll, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasVisibilityAll> _HasVisibilityAllNgFactory = const ComponentFactory('has-visibility-all', viewFactory_HasVisibilityAllHost0);
ComponentFactory<import1.HasVisibilityAll> get HasVisibilityAllNgFactory {
  return _HasVisibilityAllNgFactory;
}

final List<dynamic> styles$HasVisibilityAllHost = const [];

// Host for component HasVisibilityAll in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class _ViewHasVisibilityAllHost0 extends import8.HostView<import1.HasVisibilityAll> {
  @override
  void build() {
    // Child component HasVisibilityAll in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this.componentView = ViewHasVisibilityAll0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasVisibilityAll();
    this.initRootNode(_el_0);
  }

  @override
  dynamic injectorGetInternal(dynamic token, int nodeIndex, dynamic notFoundResult) {
    if ((identical(token, import1.HasVisibilityAll) && (0 == nodeIndex))) {
      return this.component;
    }
    return notFoundResult;
  }
}

import8.HostView<import1.HasVisibilityAll> viewFactory_HasVisibilityAllHost0() {
  return _ViewHasVisibilityAllHost0();
}

final List<dynamic> styles$HasVisibilityLocal = const [];

// View for component HasVisibilityLocal in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class ViewHasVisibilityLocal0 extends import0.ComponentView<import1.HasVisibilityLocal> {
  static import2.ComponentStyles _componentStyles;
  ViewHasVisibilityLocal0(import3.View parentView, int parentIndex) : super(parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    this.initComponentStyles();
    this.rootElement = import5.document.createElement('has-visibility-local');
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/visibility.dart' : null);
  }

  @override
  void build() {
    final import5.HtmlElement parentRenderNode = this.initViewRoot();
  }

  static void _debugClearComponentStyles() {
    _componentStyles = null;
  }

  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      _componentStyles = (styles = import2.ComponentStyles.unscoped(styles$HasVisibilityLocal, _debugComponentUrl));
      if (import6.isDevMode) {
        import2.ComponentStyles.debugOnClear(_debugClearComponentStyles);
      }
    }
    this.componentStyles = styles;
  }
}

const ComponentFactory<import1.HasVisibilityLocal> _HasVisibilityLocalNgFactory = const ComponentFactory('has-visibility-local', viewFactory_HasVisibilityLocalHost0);
ComponentFactory<import1.HasVisibilityLocal> get HasVisibilityLocalNgFactory {
  return _HasVisibilityLocalNgFactory;
}

final List<dynamic> styles$HasVisibilityLocalHost = const [];

// Host for component HasVisibilityLocal in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
class _ViewHasVisibilityLocalHost0 extends import8.HostView<import1.HasVisibilityLocal> {
  @override
  void build() {
    // Child component HasVisibilityLocal in asset:_goldens/test/_files/visibility.dart (changeDetection: Default)
    this.componentView = ViewHasVisibilityLocal0(this, 0);
    final _el_0 = this.componentView.rootElement;
    this.component = import1.HasVisibilityLocal();
    this.initRootNode(_el_0);
  }
}

import8.HostView<import1.HasVisibilityLocal> viewFactory_HasVisibilityLocalHost0() {
  return _ViewHasVisibilityLocalHost0();
}

var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;

  _ngRef.registerComponent(Dependent, DependentNgFactory);
  _ngRef.registerComponent(DependencyWithDependentInView, DependencyWithDependentInViewNgFactory);
  _ngRef.registerComponent(DependencyWithContent, DependencyWithContentNgFactory);
  _ngRef.registerComponent(DependencyAndDependentInView, DependencyAndDependentInViewNgFactory);
  _ngRef.registerComponent(HasVisibilityAll, HasVisibilityAllNgFactory);
  _ngRef.registerComponent(HasVisibilityLocal, HasVisibilityLocalNgFactory);
  _ref0.initReflector();
}
